sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,776 @@
+            List of n_features-dimensional data points. Each row
+            corresponds to a single data point.
+
+        Returns
+        -------
+        self : object
+            Returns an instance of self.
+        """
+
+        self._check_parameters()
+        X = check_array(
+            X, dtype=[np.float64, np.float32], ensure_min_samples=1
+        )
+
+        # check n_components against sample size
+        n_comps = [self.max_components, self.min_components]
+        names = ["max_components", "min_components"]
+        for i in range(len(names)):
+            if n_comps[i] > X.shape[0]:
+                msg = names[i] + "must be <= n_samples, but" + names[i]
+                msg += "= {}, n_samples = {}".format(n_comps[i], X.shape[0])
+                raise ValueError(msg)
+
+        # check if X contains the 0 vector
+        if np.any(~X.any(axis=1)):
+            if "cosine" in self.affinity or self.affinity == "all":
+                if isinstance(self.affinity, np.ndarray):
+                    self.affinity = np.delete(
+                        self.affinity, np.argwhere(self.affinity == "cosine")
+                    )
+                if isinstance(self.affinity, list):
+                    self.affinity.remove("cosine")
+                if self.affinity == "all":
+                    self.affinity = ["euclidean", "manhattan", "none"]
+                warnings.warn(
+                    "X contains a zero vector, will not run cosine affinity."
+                )
+            if self.affinity == "cosine":
+                msg = "X contains a zero vector, cannot run cosine affinity."
+                raise ValueError(msg)
+
+        label_init = self.label_init
+        if label_init is not None:
+            if label_init.size != X.shape[0]:
+                msg = "n_samples must be the same as the length of label_init"
+                raise ValueError(msg)
+
+        param_grid = dict(
+            n_components=range(self.min_components, self.max_components + 1),
+            random_state=[self.random_state],
+        )
+        param_grid["affinity"] = (
+            self.affinity
+            if self.affinity != "all"
+            else ["euclidean", "manhattan", "cosine", "none"]
+        )
+        param_grid["linkage"] = (
+            self.linkage
+            if self.linkage != "all"
+            else ["ward", "complete", "average", "single"]
+        )
+        param_grid["covariance_type"] = (
+            self.covariance_type
+            if self.covariance_type != "all"
+            else ["spherical", "diag", "tied", "full"]
+        )
+
+        param_grid = list(ParameterGrid(param_grid))
+        param_grid_ag, param_grid = _process_paramgrid(param_grid)
+
+        n = X.shape[0]
+        if self.max_agglom_size is None or n <= self.max_agglom_size:
+            X_subset = X
+        else:  # if dataset is huge, agglomerate a subset
+            subset_idxs = np.random.choice(
```

Comment:
@tliu68 [randomness] this does not use the random state which may be causing the issue we talked about

Reply:
though, I doubt this caused the actual specific test failure

Reply:
In this case, one should use: 
```suggestion
            subset_idxs = self.random_state.choice(
```